uaclient/contract.py

```diff
@@ -153,25 +153,34 @@ def request_resource_machine_access(
         return resource_access
 
     def request_machine_token_update(
-        self, machine_token, contract_id, machine_id=None
-    ):
+        self,
+        machine_token: str,
+        contract_id: str,
+        machine_id: str = None,
+        detach: bool = False,
```

Comment:
This feels like a strange public interface for us to have on this class. This method will now behave in two very different ways depending on the value of a parameter. Perhaps, as a rough proposal:

```py
def detach_machine_from_contract(self, ...):
    self._request_machine_token_update(..., detach=True)

def request_machine_token_update(self, ...):
    self._request_machine_token_update(..., detach=False)

def _request_machine_token_update(self, ...):
    <current contents of request_machine_token_update>
```

(I think using the same method for the internal implementation is fine, to be clear. This is just about making the public interface easier to understand.)